In [1]:
import numpy as np 
import pandas as pd
import os 
import sys 
import subprocess
if sys.platform == 'darwin':
    print("Current system is macOS")
    main_fold_path = '/Users/shanxiafeng/Documents/Project/Research/fnirs-prognosis/code/fnirs-treatment-response-prediction'
elif sys.platform == 'linux':
    print("Current system is Ubuntu")
    main_fold_path = '/home/jy/Documents/fnirs/treatment_response/fnirs-depression-deeplearning'
else:
    print("Current system is neither macOS nor Ubuntu")
os.chdir(main_fold_path)
from utils.fnirs_utils import print_md_table_val_test_AUC


Current system is Ubuntu


In [ ]:


for itr in range(4, 14):
    bash_code = f"conda run -n tf python scripts/plot/DL/read_LOO_nestedCV_gnntr_auc.py --max 4 --K_FOLD 5 --result_path results/gnn_transformer/prognosis_mix_hb/pretreatment_response/loocv_v{itr}l1_rate_0.01_l2_rate_0.001_d_model_16_batch_size_64_n_layers_6 "
    subprocess.run(bash_code, shell=True)



Average Test Metrics:
Accuracy: 0.6328571428571428 (±0.03142135559299737)
Sensitivity: 0.43571428571428567 (±0.05270462766947301)
Specificity: 0.8299999999999998 (±0.019436506316150997)
AUC: 0.5992857142857143 (±0.04702606986731438)

Average Validation Metrics:
Accuracy: 0.8298838578125004 (±0.012694089454970334)
Sensitivity: 0.7053125000000005 (±0.023664410837326493)
Specificity: 0.9544552156250002 (±0.003440104679139837)
AUC: 0.829882626645267 (±0.01269407719856165)
| Model Name | Testing Set |             |             |             | Validation Set |             |             |             |
|------------|-------------|-------------|-------------|-------------|-------------|-------------|-------------|-------------|
|            | Balanced Accuracy | Sensitivity | Specificity | AUC | Balanced Accuracy | Sensitivity | Specificity | AUC |
| Mean   | 63.2857  | 43.5714  | 83.0000  | 59.9286  | 82.9884  | 70.5313  | 95.4455  | 82.9883  |

| Metric         | Test Set Average | Validation Set Average |
|----------------|------------------|------------------------|
| Accuracy       | 0.7437 | 0.9101 |
| Sensitivity    | 0.4358 | 0.7063 |
| Specificity    | 0.8320 | 0.9552 |
| AUC            | 0.5993 | 0.8327 |


In [9]:
csv_path = 'results/gnn_transformer/prognosis_mix_hb/metrics.txt'

df = pd.read_csv(csv_path)
import pandas as pd

# Assuming 'df' is your DataFrame
# Calculating the balanced accuracy for each metric in the test set
balanced_test_accuracy = (df['Test Sensitivity'] + df['Test Specificity']) / 2

# Calculating the balanced accuracy for each metric in the validation set
balanced_validation_accuracy = (df['Validation Sensitivity'] + df['Validation Specificity']) / 2

df['Test Accuracy'] = balanced_test_accuracy
df['Validation Accuracy'] = balanced_validation_accuracy


# Calculating the mean and standard deviation for each metric in the test set
average_test_accuracy = df['Test Accuracy'].mean()
std_test_accuracy = df['Test Accuracy'].std()

average_test_sensitivity = df['Test Sensitivity'].mean()
std_test_sensitivity = df['Test Sensitivity'].std()

average_test_specificity = df['Test Specificity'].mean()
std_test_specificity = df['Test Specificity'].std()

average_test_auc = df['Test AUC'].mean()
std_test_auc = df['Test AUC'].std()

# Calculating the mean and standard deviation for each metric in the validation set
average_validation_accuracy = df['Validation Accuracy'].mean()
std_validation_accuracy = df['Validation Accuracy'].std()

average_validation_sensitivity = df['Validation Sensitivity'].mean()
std_validation_sensitivity = df['Validation Sensitivity'].std()

average_validation_specificity = df['Validation Specificity'].mean()
std_validation_specificity = df['Validation Specificity'].std()

average_validation_auc = df['Validation AUC'].mean()
std_validation_auc = df['Validation AUC'].std()

# Printing the results with mean and standard deviation
print("Average Test Metrics:")
print(f"Accuracy: {average_test_accuracy} (±{std_test_accuracy})")
print(f"Sensitivity: {average_test_sensitivity} (±{std_test_sensitivity})")
print(f"Specificity: {average_test_specificity} (±{std_test_specificity})")
print(f"AUC: {average_test_auc} (±{std_test_auc})")

print("\nAverage Validation Metrics:")
print(f"Accuracy: {average_validation_accuracy} (±{std_validation_accuracy})")
print(f"Sensitivity: {average_validation_sensitivity} (±{std_validation_sensitivity})")
print(f"Specificity: {average_validation_specificity} (±{std_validation_specificity})")
print(f"AUC: {average_validation_auc} (±{std_validation_auc})")

mean_test_metrics = [average_test_accuracy, average_test_sensitivity, average_test_specificity, average_test_auc]
mean_val_metrics = [average_validation_accuracy, average_validation_sensitivity, average_validation_specificity, average_validation_auc]

std_test_metrics = [std_test_accuracy, std_test_sensitivity, std_test_specificity, std_test_auc]
std_val_metrics = [std_validation_accuracy, std_validation_sensitivity, std_validation_specificity, std_validation_auc]



print_md_table_val_test_AUC('Mean', mean_test_metrics, mean_val_metrics, already_balanced_accuracy=True)
print_md_table_val_test_AUC('STD', std_test_metrics, std_val_metrics, already_balanced_accuracy=True, print_table_header=False)


Average Test Metrics:
Accuracy: 0.6328571428571428 (±0.030583301748906402)
Sensitivity: 0.4357142857142858 (±0.05129891760425773)
Specificity: 0.8300000000000001 (±0.01891810605853834)
AUC: 0.5992857142857143 (±0.045771815304421606)

Average Validation Metrics:
Accuracy: 0.8298838578125004 (±0.012355519388077888)
Sensitivity: 0.7053125000000003 (±0.023033246137519933)
Specificity: 0.9544552156250005 (±0.0033483520193319017)
AUC: 0.829882626645267 (±0.012355507458565678)
| Model Name | Testing Set |             |             |             | Validation Set |             |             |             |
|------------|-------------|-------------|-------------|-------------|-------------|-------------|-------------|-------------|
|            | Balanced Accuracy | Sensitivity | Specificity | AUC | Balanced Accuracy | Sensitivity | Specificity | AUC |
| Mean   | 63.2857  | 43.5714  | 83.0000  | 59.9286  | 82.9884  | 70.5313  | 95.4455  | 82.9883  |
| STD   | 3.0583  | 5.1299  | 1.8918  | 4.5772